src/Native/CpuMathNative/Sse.cpp

```diff
@@ -266,7 +266,8 @@ EXPORT_API(void) Scale(float a, _Inout_ float * pd, int c)
             _mm_storeu_ps(pd, result);
 
             pd += misalignment;
-            c -= misalignment;
+			// safe to downcast as misalignment < 16.
```

Comment:
I am not sure if it's still 16 for AVX or longer vector instruction sets, but I believe it's safe.

Reply:
true, there is  >>=2 there.  It is within 128 then

---
In reply to: [276744696](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276744696) [](ancestors = 276744696)